### PR TITLE
add repair PBO, fix #185

### LINKED ISSUE #185: Repairing vehicle hitpoints

### PR DESCRIPTION
**Pull request by commy2**
- [x] Replacing wheels logic
- [x] Tracks
- [x] Localizations

- [x] Toolkit, engineer, disable default repair
- [x] progress bar abort conditions (correct wheel is in nearest objects)

### CODE REVIEW

**Comment by AlexSegen** [comment]
hallo, may I know status check?

**Comment by Glowbal** [comment]
This is now functional from what I can tell. A few things; 
* I couldn't get a proper selection position for tracks, so tracks are still treated as regular hitpoints. 
* On top of that we don't seem to be able to remove them either - they stick on the model (atleast the vanilla ones I tried).
* I've disabled the vanilla repair actions with this as well. We could implement a disable toggle and use `addAction` to renable the repair functionality using `setDamage` if we so desire.
* The logistics module is probably going to be a requirement for this, since we want to be able to load/unload wheels.

I think it would be good if someone else could take a look at the code and features and provide feedback.

**Comment by Glowbal** [comment]
This is ready for testing

File: addons/repair/functions/fnc_moduleAssignEngineer.sqf [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            if (_nilCheckPassedList == "") then {
+                _nilCheckPassedList = _x;
+            } else {
+                _nilCheckPassedList = _nilCheckPassedList + ","+ _x;
+            };
+        };
+    }foreach _splittedList;
+
+    _list = "[" + _nilCheckPassedList + "]";
+    _parsedList = [] call compile _list;
```
Comment (jonpas) [inline addons/repair/functions/fnc_moduleAssignEngineer.sqf]: L22 - L38 is being used in every module assign function, I am also using it in slideshow, I think it warrants a function in common.

File: addons/repair/functions/fnc_moduleAssignEngineer.sqf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    _splittedList = [_list, ","] call BIS_fnc_splitString;
+    _nilCheckPassedList = "";
+    {
+        _x = [_x] call EFUNC(common,stringRemoveWhiteSpace);
+        if !(isnil _x) then {
+            if (_nilCheckPassedList == "") then {
+                _nilCheckPassedList = _x;
+            } else {
+                _nilCheckPassedList = _nilCheckPassedList + ","+ _x;
```
Comment (jokoho48) [inline addons/repair/functions/fnc_moduleAssignEngineer.sqf]: better use: `_nilCheckPassedList = format ["%1,%2",_nilCheckPassedList, _x];`

File: addons/repair/functions/fnc_moduleAssignEngineer.sqf [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            };
+        };
+    }foreach _splittedList;
+
+    _list = "[" + _nilCheckPassedList + "]";
```
Comment (jokoho48) [inline addons/repair/functions/fnc_moduleAssignEngineer.sqf]: better use `_list = format["[%1]",_nilCheckPassedList];`

**Pull request merged**